java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/Database.java

```diff
@@ -0,0 +1,290 @@
+            throws SQLException {
+        String query = "SELECT DISTINCT(collection) FROM " + TABLE_NAME;
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String collection = resultSet.getString("collection");
+                if (collectionName.equals(collection)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public Mono<List<String>> getCollectionsAsync(Connection connection) {
+        return Mono.defer(
+                        () -> {
+                            List<String> collections = new ArrayList<>();
+                            String query = "SELECT DISTINCT(collection) FROM " + TABLE_NAME;
+                            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                                ResultSet resultSet = statement.executeQuery();
+                                while (resultSet.next()) {
+                                    String collection = resultSet.getString("collection");
+                                    collections.add(collection);
+                                }
+                            } catch (SQLException e) {
+                                throw new RuntimeException("Failed to get collections", e);
+                            }
+                            return Mono.just(collections);
+                        })
+                .subscribeOn(Schedulers.boundedElastic());
+    }
+
+    public Mono<List<DatabaseEntry>> readAllAsync(Connection connection, String collectionName) {
+        return Mono.defer(
+                        () -> {
+                            List<DatabaseEntry> entries = new ArrayList<>();
+                            String query = "SELECT * FROM " + TABLE_NAME + " WHERE collection = ?";
+                            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                                statement.setString(1, collectionName);
+                                ResultSet resultSet = statement.executeQuery();
+                                while (resultSet.next()) {
+                                    String key = resultSet.getString("key");
+                                    String metadata = resultSet.getString("metadata");
+                                    String embedding = resultSet.getString("embedding");
+                                    String timestamp = resultSet.getString("timestamp");
+                                    entries.add(
+                                            new DatabaseEntry(key, metadata, embedding, timestamp));
+                                }
+                            } catch (SQLException e) {
+                                throw new RuntimeException("Failed to read all entries", e);
```

Comment:
should be Mono.error, and something more specific like creating a StorageConnectorException